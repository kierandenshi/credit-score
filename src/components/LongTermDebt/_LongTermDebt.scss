@import '~@css/includes/helpers';

.long-term-debt {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 298px;
  height: 298px;
  
  @include media('>phone') {
    width: 414px;
    height: 414px;
  }

  &__inner {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 320px;
    height: 320px;
  }
}
