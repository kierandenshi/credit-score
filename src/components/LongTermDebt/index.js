export { default } from './LongTermDebt.jsx';
