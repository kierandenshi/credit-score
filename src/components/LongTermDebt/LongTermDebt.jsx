import React from 'react';
import PropTypes from 'prop-types';
import './_LongTermDebt.scss';

const LongTermDebt = ({ data }) => {
    const { currentLongTermDebt } = data;

    return (
        <div className={'long-term-debt'}>
            {currentLongTermDebt}
        </div>
    );
};

LongTermDebt.propTypes = {
    data: PropTypes.shape({}).isRequired,
};

export default LongTermDebt;
