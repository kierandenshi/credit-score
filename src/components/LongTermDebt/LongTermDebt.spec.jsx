import React from 'react';
import { shallow } from 'enzyme';
import { getProps } from '@test/helpers';
import LongTermDebt from './LongTermDebt.jsx';

const initialProps = {
    data: { currentLongTermDebt: 253400 },
};

describe('CreditRating component', () => {
    const wrapper = shallow(<LongTermDebt {...getProps(initialProps)} />);

    it('should render', () => {
        expect(wrapper.find('.long-term-debt')).toHaveLength(1);
    });
});
