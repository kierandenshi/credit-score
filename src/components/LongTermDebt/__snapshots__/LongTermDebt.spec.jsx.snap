// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditRating component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <LongTermDebt
      data={
        Object {
          "currentLongTermDebt": 253400,
        }
      }
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="long-term-debt__info"
        >
          <span>
            Your long term debt total
          </span>
          <span
            className="strong"
          >
            £253,400.00
          </span>
          <span>
            Total credit limit 0
          </span>
        </div>,
        "className": "long-term-debt",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Your long term debt total
            </span>,
            <span
              className="strong"
            >
              £253,400.00
            </span>,
            <span>
              Total credit limit 0
            </span>,
          ],
          "className": "long-term-debt__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your long term debt total",
            },
            "ref": null,
            "rendered": "Your long term debt total",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "£253,400.00",
              "className": "strong",
            },
            "ref": null,
            "rendered": "£253,400.00",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Total credit limit 0",
            },
            "ref": null,
            "rendered": "Total credit limit 0",
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="long-term-debt__info"
          >
            <span>
              Your long term debt total
            </span>
            <span
              className="strong"
            >
              £253,400.00
            </span>
            <span>
              Total credit limit 0
            </span>
          </div>,
          "className": "long-term-debt",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Your long term debt total
              </span>,
              <span
                className="strong"
              >
                £253,400.00
              </span>,
              <span>
                Total credit limit 0
              </span>,
            ],
            "className": "long-term-debt__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your long term debt total",
              },
              "ref": null,
              "rendered": "Your long term debt total",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "£253,400.00",
                "className": "strong",
              },
              "ref": null,
              "rendered": "£253,400.00",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total credit limit 0",
              },
              "ref": null,
              "rendered": "Total credit limit 0",
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="long-term-debt__info"
      >
        <span>
          Your long term debt total
        </span>
        <span
          className="strong"
        >
          £253,400.00
        </span>
        <span>
          Total credit limit 0
        </span>
      </div>,
      "className": "long-term-debt",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Your long term debt total
          </span>,
          <span
            className="strong"
          >
            £253,400.00
          </span>,
          <span>
            Total credit limit 0
          </span>,
        ],
        "className": "long-term-debt__info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your long term debt total",
          },
          "ref": null,
          "rendered": "Your long term debt total",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "£253,400.00",
            "className": "strong",
          },
          "ref": null,
          "rendered": "£253,400.00",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Total credit limit 0",
          },
          "ref": null,
          "rendered": "Total credit limit 0",
          "type": "span",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="long-term-debt__info"
        >
          <span>
            Your long term debt total
          </span>
          <span
            className="strong"
          >
            £253,400.00
          </span>
          <span>
            Total credit limit 0
          </span>
        </div>,
        "className": "long-term-debt",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Your long term debt total
            </span>,
            <span
              className="strong"
            >
              £253,400.00
            </span>,
            <span>
              Total credit limit 0
            </span>,
          ],
          "className": "long-term-debt__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your long term debt total",
            },
            "ref": null,
            "rendered": "Your long term debt total",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "£253,400.00",
              "className": "strong",
            },
            "ref": null,
            "rendered": "£253,400.00",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Total credit limit 0",
            },
            "ref": null,
            "rendered": "Total credit limit 0",
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
